bumped istio to 1.4.5 according to the docs to fix issue #25

aidun opened the pull request:
bumped istio to 1.4.5 according to the docs to fix issue #25

## Description
bumped istio to 1.4.5 according to the docs to fix issue #25

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))


## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
kind create cluster
arkade install istio
helm list -A
kubectl get all -n Istio-system

--> processes are running
--> helm charts in 1.4.5 are present

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

aidun commented:
Need x86 test k8s to test (only have rapis currently). Maybe someone can confirm that it worked.

aidun commented:
Testet it with Kind. Is working as expected.

alexellis commented on cmd/apps/istio_app.go:
```diff
@@ -114,11 +114,21 @@ func MakeInstallIstio() *cobra.Command {
 				return fmt.Errorf("unable to istio-init install chart with helm %s", err)
 			}
 
+			err = kubectl("-n", "istio-system", "wait", "--for=condition=complete", "job", "--all")
```
Why is this required? Isn't it part of the `--wait` flag that is optional on apps like `openfaas` and `inlets-operator`?

aidun replied:
This are two different points.

I will remove the `kubectl` command, it is not needed, but we have to force wait here, if not the second install step will fail. Is it okay for you?

alexellis commented on cmd/apps/istio_app.go:
```diff
@@ -114,11 +114,21 @@ func MakeInstallIstio() *cobra.Command {
+			err = kubectl("-n", "istio-system", "wait", "--for=condition=complete", "job", "--all")
+
+			if err != nil {
+				return fmt.Errorf("error waiting for the crds: %s", err)
```
Should be: `CRDs`

aidun replied:
Done

alexellis commented on cmd/apps/istio_app.go:
```diff
@@ -114,11 +114,21 @@ func MakeInstallIstio() *cobra.Command {
+			}
+		}
+
+		if err != nil {
```
Why do we have the same error twice, is this a different step?

aidun replied:
i removed it. copy paste error.

alexellis commented on cmd/apps/istio_app.go:
```diff
@@ -114,11 +114,21 @@ func MakeInstallIstio() *cobra.Command {
 		if initIstio, _ := command.Flags().GetBool("init"); initIstio {
-			err = helm3Upgrade(outputPath, "istio/istio-init", namespace, "", defaultVersion, overrides, wait)
+			// Waiting for the crds to appear
+			err = helm3Upgrade(outputPath, "istio/istio-init", namespace, "", defaultVersion, overrides, true)
```
Please use the wait flag instead of forcing wait to be true.

I'm OK if you want to add a flag and set it to default a true value. See the other apps for examples or ask in #arkade on Slack if you need help with that.

aidun replied:
If I don´t force and to wait, the second step will always fail. I think its a bad behavior.

aidun commented:
Thank you for the feedback. I followed your comments and fixed the most of it. I have only concerns to implement the optional wait on the first step. I added my explanation to your inline review.
